### PR TITLE
[FLINK-26717][fs]Move s3 common utils to flink-core

### PR DESCRIPTION
**Pull request by wujinhu**
## What is the purpose of the change

Utility classes are common and can be used by other cloud storages(oss) in flink-s3-fs-base module. I will move them to flink-core module like RefCountedFile.

## Brief change log

**- move utility classes from flink-s3-fs-base to flink-core**

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by JingsongLi** [comment]
@flinkbot run azure

**Comment by JingsongLi** [comment]
Results:
2022-03-18T10:30:34.2107707Z Mar 18 10:30:34 [INFO] 
2022-03-18T10:30:34.2108281Z Mar 18 10:30:34 [ERROR] Failures: 
2022-03-18T10:30:34.2110167Z Mar 18 10:30:34 [ERROR]   Architecture Violation [Priority: MEDIUM] - Rule 'Production code must not call methods annotated with @VisibleForTesting' was violated (1 times):
2022-03-18T10:30:34.2111733Z Mar 18 10:30:34 Method <org.apache.flink.core.fs.RefCountedBufferingFileStream.toString()> calls method <org.apache.flink.core.fs.RefCountedFileWithStream.getReferenceCounter()> in (RefCountedBufferingFileStream.java:154)

**Comment by MartijnVisser** [comment]
Like I said in the Jira, I'm not sure this is a good idea because there's also a plan to externalize filesystem implementations to decouple them from Flink releases. But I would like to get @AHeise his opinion on this.

File: flink-core/src/main/java/org/apache/flink/core/fs/RefCountedBufferingFileStream.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * limitations under the License.
  */
 
-package org.apache.flink.fs.s3.common.utils;
+package org.apache.flink.core.fs;
```
Comment (JingsongLi) [inline flink-core/src/main/java/org/apache/flink/core/fs/RefCountedBufferingFileStream.java:19]: Need to update `RefCountedBufferingFileStream.sync`

Comment (wujinhu) [reply flink-core/src/main/java/org/apache/flink/core/fs/RefCountedBufferingFileStream.java:19]: OK

**Comment by wujinhu** [comment]
Hi @JingsongLi , I upload a patch to update the pr and please take a look.

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**